src/api/factories/ItemCategoryFactory.ts

```diff
@@ -34,58 +42,14 @@ export class ItemCategoryFactory {
 
     /**
      *
-     * @param categoryAsArray
-     * @param rootCategoryWithRelated
-     * @returns {Promise<any>}
+     * @param category : ItemCategoryMessage
+     * @returns {Promise<ItemCategoryCreateRequest>}
      */
-    public async getModel(categoryAsArray: string[], rootCategoryWithRelated: ItemCategory): Promise<ItemCategory> {
-        const rootCategory: any = rootCategoryWithRelated;
-        let childItemCategories;
-        const createdCategories: any = [];
-        let findItemCategory;
-        let lastCheckIndex = 0;
-        // check cat1 match with root itemcategory.key
-        if (categoryAsArray[0] !== rootCategory.key) { // cat_ROOT
-            this.log.warn(`${categoryAsArray[0]} should be root ItemCategory`);
-        }
-        // insert root category
-        createdCategories.push({
-            parentCategoryId: null,
-            id: rootCategory.id
-        });
-        childItemCategories = rootCategory.ChildItemCategories;
-        for (let c = 1; c <= categoryAsArray.length; c++) {
-            // check category have ChildItemCategories
-            if (childItemCategories.length > 0) {
-                // search catgeory
-                findItemCategory = await this.checkCategory(childItemCategories, categoryAsArray[c]);
-                if (findItemCategory) {
-                    createdCategories.push({
-                        parentCategoryId: findItemCategory.parentItemCategoryId,
-                        id: findItemCategory.id
-                    });
-                    childItemCategories = findItemCategory.ChildItemCategories || [];
-                } else {
-                    // created all category till from
-                    break;
-                }
-            } else {
-                // created all category till from
-                break;
-            }
-            lastCheckIndex = c;
-        }
-        const ItemCategoryOutput = {
-            lastCheckIndex, // created all category till that index
-            createdCategories
-        };
-        return ItemCategoryOutput as any;
-    }
-
-    private async checkCategory(categories: string[], value: string): Promise<any> {
-        return _.find(categories, (itemcategory) => {
-            return (itemcategory['key'] === value || itemcategory['name'] === value);
-        });
+    public async getModel(category: ItemCategoryMessage): Promise<ItemCategoryCreateRequest> {
+        return {
```

Comment:
just pass the name and parent_id, theres no ItemCategoryMessage that we are receiving from the network.